project/CompilerSettings.scala

```diff
@@ -66,6 +66,9 @@ object CompilerSettings {
         )
       )
     },
-    scalacOptions in (Test, console) := (scalacOptions in (Compile, console)).value
+    scalacOptions in (Test, console) := (scalacOptions in (Compile, console)).value,
+    // To prevent "dead code following this construct" error when using * mockito-scala library
+    // https://github.com/mockito/mockito-scala/#dead-code-warning
+    scalacOptions in Test -= "-Ywarn-dead-code"
```

Comment:
I cannot compile the project in IntelliJ because of _dead code_ error in tests.

I found this old bug which would explain the behavior. I noticed that scalaOptions set for Compile and Test configuration are loaded differently in IntelliJ. 
https://youtrack.jetbrains.com/issue/SCL-11824

I'll try to load separately options for Compile and Test configuration without using `-=` which doesn't have effect in IntelliJ.

Reply:
I couldn't find a workaround, it seems that IntelliJ cannot use separate configuration for Test scope. This is another related issue I found. 
https://youtrack.jetbrains.com/issue/IDEA-232043

@stanislavlyalin Let's disable dead code detection completely, this seems the only solution for now. Please add this additional information in the comments also.

Reply:
I [returned](https://github.com/rchain/rchain/pull/3741/commits/871a7bd0580ca63944339224581797b06ba31d43) initial version of my code and added some related info.